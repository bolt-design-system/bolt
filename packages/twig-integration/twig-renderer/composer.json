{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "description": "Twig rendering service used by the Bolt Design System.",
  "version": "2.20.0",
  "license": "MIT",
  "repositories": [
    {
      "type": "path",
	    "url": "../twig-extensions-shared"
	  },
    {
      "type": "path",
      "url": "../twig-extensions-compat"
    }
  ],
  "require": {
    "bolt-design-system/core-php": "*",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.42.5",
    "webmozart/path-util": "*",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
