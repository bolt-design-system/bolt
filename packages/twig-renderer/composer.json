{
  "name": "bolt-design-system/twig-renderer",
  "type": "library",
  "version": "2.3.0",
  "license": "MIT",
  "require": {
    "bolt-design-system/core-php": "^2.3.0",
    "bolt-design-system/drupal-twig-extensions": "*",
    "twig/twig": "^1.38.2",
    "webmozart/path-util": "^2.3",
    "symfony/finder": "*"
  },
  "scripts": {
    "setup": "@composer install --no-interaction --prefer-dist --no-progress"
  }
}
