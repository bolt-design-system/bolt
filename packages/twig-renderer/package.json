{
  "name": "@bolt/twig-renderer",
  "version": "2.7.0",
  "license": "MIT",
  "description": "Renders Twig templates using an on-the-fly PHP rendering service.",
  "main": "twig-renderer.js",
  "repository": "https://github.com/bolt-design-system/bolt/tree/master/packages/twig-renderer",
  "keywords": [
    "design system",
    "bolt design system",
    "twig",
    "php"
  ],
  "maintainers": [
    {
      "name": "Salem Ghoweri",
      "email": "me@salemghoweri.com",
      "web": "https://github.com/sghoweri"
    }
  ],
  "dependencies": {
    "@basalt/twig-renderer": "^0.12.1",
    "@bolt/build-utils": "^2.7.0",
    "@bolt/drupal-twig-extensions": "^2.7.0",
    "sleep-promise": "^8.0.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "setup": "composer run setup"
  },
  "schema": "utils/config.schema.yml",
  "bugs": "https://github.com/bolt-design-system/bolt/issues",
  "homepage": "https://boltdesignsystem.com",
  "gitHead": "3d198feea5f73c63913a956a2a077e229ad8d462"
}
