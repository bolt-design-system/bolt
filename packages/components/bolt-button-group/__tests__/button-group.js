import { render } from '@bolt/twig-renderer';

describe('<bolt-button-group> Component', () => {
  test('basic usage', async () => {
    const results = await render(
      '@bolt-components-button-group/button-group.twig',
      {
        buttons: [
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'About Pega',
            style: 'secondary',
          },
        ],
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('icon only button group', async () => {
    const results = await render(
      '@bolt-components-button-group/button-group.twig',
      {
        buttons: [
          {
            text: 'Example',
            style: 'primary',
            size: 'xsmall',
            iconOnly: true,
            icon: {
              name: 'close',
            },
            border_radius: 'full',
          },
          {
            text: 'About Pega',
            style: 'secondary',
            size: 'xsmall',
            iconOnly: true,
            icon: {
              name: 'close',
            },
          },
        ],
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('button group quantity 3', async () => {
    const results = await render(
      '@bolt-components-button-group/button-group.twig',
      {
        buttons: [
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
        ],
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('button group quantity 4', async () => {
    const results = await render(
      '@bolt-components-button-group/button-group.twig',
      {
        buttons: [
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
        ],
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('button group quantity 5', async () => {
    const results = await render(
      '@bolt-components-button-group/button-group.twig',
      {
        buttons: [
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
          {
            text: 'Learn More',
            style: 'primary',
          },
        ],
      },
    );
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
