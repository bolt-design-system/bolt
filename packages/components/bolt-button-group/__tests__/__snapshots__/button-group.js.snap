// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-button-group> Component basic usage 1`] = `
<bolt-button-group bolt-component>
  <ul class="c-bolt-button-group">
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="secondary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--secondary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            About Pega
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
  </ul>
</bolt-button-group>
`;

exports[`<bolt-button-group> Component button group quantity 3 1`] = `
<bolt-button-group bolt-component>
  <ul class="c-bolt-button-group">
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
  </ul>
</bolt-button-group>
`;

exports[`<bolt-button-group> Component button group quantity 4 1`] = `
<bolt-button-group bolt-component>
  <ul class="c-bolt-button-group">
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
  </ul>
</bolt-button-group>
`;

exports[`<bolt-button-group> Component button group quantity 5 1`] = `
<bolt-button-group bolt-component>
  <ul class="c-bolt-button-group">
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="medium"
                   color="primary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
      >
        <button class="c-bolt-button c-bolt-button--medium c-bolt-button--primary c-bolt-button--border-radius-regular c-bolt-button--center"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Learn More
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
  </ul>
</bolt-button-group>
`;

exports[`<bolt-button-group> Component content items button group 1`] = `
<bolt-button-group bolt-component>
  <ul class="c-bolt-button-group">
  </ul>
  <bolt-button size="xsmall"
               color="primary"
               width="auto"
               border-radius="full"
               align="center"
               transform="none"
               icon-only
  >
    <button class="c-bolt-button c-bolt-button--xsmall c-bolt-button--primary c-bolt-button--border-radius-full c-bolt-button--center c-bolt-button--icon-only"
            is="shadow-root"
    >
      <replace-with-children class="c-bolt-button__item">
        Example
      </replace-with-children>
      <replace-with-children class="c-bolt-button__icon c-bolt-button__icon-sizer">
        <bolt-icon slot="after"
                   name="close"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.414 11.998l5.296-5.294a1.005 1.005 0 000-1.413 1.005 1.005 0 00-1.414 0L12 10.585 6.704 5.291a1.006 1.006 0 00-1.413 0 1.004 1.004 0 000 1.413l5.296 5.294-5.296 5.295a.999.999 0 101.413 1.413l5.295-5.295 5.296 5.295a.999.999 0 001.413 0 1.005 1.005 0 000-1.413l-5.296-5.295h.002z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </replace-with-children>
    </button>
  </bolt-button>
  <bolt-button size="xsmall"
               color="secondary"
               width="auto"
               border-radius="regular"
               align="center"
               transform="none"
               icon-only
  >
    <button class="c-bolt-button c-bolt-button--xsmall c-bolt-button--secondary c-bolt-button--border-radius-regular c-bolt-button--center c-bolt-button--icon-only"
            is="shadow-root"
    >
      <replace-with-children class="c-bolt-button__item">
        About Pega
      </replace-with-children>
      <replace-with-children class="c-bolt-button__icon c-bolt-button__icon-sizer">
        <bolt-icon slot="after"
                   name="close"
        >
          <span class="c-bolt-icon c-bolt-icon--close">
            <svg class="c-bolt-icon__icon"
                 xmlns="http://www.w3.org/2000/svg"
                 viewbox="0 0 24 24"
            >
              <path fill="var(--bolt-theme-icon, currentColor)"
                    fill-rule="evenodd"
                    d="M13.414 11.998l5.296-5.294a1.005 1.005 0 000-1.413 1.005 1.005 0 00-1.414 0L12 10.585 6.704 5.291a1.006 1.006 0 00-1.413 0 1.004 1.004 0 000 1.413l5.296 5.294-5.296 5.295a.999.999 0 101.413 1.413l5.295-5.295 5.296 5.295a.999.999 0 001.413 0 1.005 1.005 0 000-1.413l-5.296-5.295h.002z"
              >
              </path>
            </svg>
          </span>
        </bolt-icon>
      </replace-with-children>
    </button>
  </bolt-button>
  <p class="c-bolt-text c-bolt-text--regular c-bolt-text--normal c-bolt-text--medium">
    This is text with the tags stripped.
  </p>
</bolt-button-group>
`;

exports[`<bolt-button-group> Component icon only button group 1`] = `
<bolt-button-group bolt-component>
  <ul class="c-bolt-button-group">
    <li class="c-bolt-button-group__item">
      <bolt-button size="xsmall"
                   color="primary"
                   width="auto"
                   border-radius="full"
                   align="center"
                   transform="none"
                   icon-only
      >
        <button class="c-bolt-button c-bolt-button--xsmall c-bolt-button--primary c-bolt-button--border-radius-full c-bolt-button--center c-bolt-button--icon-only"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            Example
          </replace-with-children>
          <replace-with-children class="c-bolt-button__icon c-bolt-button__icon-sizer">
            <bolt-icon slot="after"
                       name="close"
            >
              <span class="c-bolt-icon c-bolt-icon--close">
                <svg class="c-bolt-icon__icon"
                     xmlns="http://www.w3.org/2000/svg"
                     viewbox="0 0 24 24"
                >
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        fill-rule="evenodd"
                        d="M13.414 11.998l5.296-5.294a1.005 1.005 0 000-1.413 1.005 1.005 0 00-1.414 0L12 10.585 6.704 5.291a1.006 1.006 0 00-1.413 0 1.004 1.004 0 000 1.413l5.296 5.294-5.296 5.295a.999.999 0 101.413 1.413l5.295-5.295 5.296 5.295a.999.999 0 001.413 0 1.005 1.005 0 000-1.413l-5.296-5.295h.002z"
                  >
                  </path>
                </svg>
              </span>
            </bolt-icon>
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
    <li class="c-bolt-button-group__item">
      <bolt-button size="xsmall"
                   color="secondary"
                   width="auto"
                   border-radius="regular"
                   align="center"
                   transform="none"
                   icon-only
      >
        <button class="c-bolt-button c-bolt-button--xsmall c-bolt-button--secondary c-bolt-button--border-radius-regular c-bolt-button--center c-bolt-button--icon-only"
                is="shadow-root"
        >
          <replace-with-children class="c-bolt-button__item">
            About Pega
          </replace-with-children>
          <replace-with-children class="c-bolt-button__icon c-bolt-button__icon-sizer">
            <bolt-icon slot="after"
                       name="close"
            >
              <span class="c-bolt-icon c-bolt-icon--close">
                <svg class="c-bolt-icon__icon"
                     xmlns="http://www.w3.org/2000/svg"
                     viewbox="0 0 24 24"
                >
                  <path fill="var(--bolt-theme-icon, currentColor)"
                        fill-rule="evenodd"
                        d="M13.414 11.998l5.296-5.294a1.005 1.005 0 000-1.413 1.005 1.005 0 00-1.414 0L12 10.585 6.704 5.291a1.006 1.006 0 00-1.413 0 1.004 1.004 0 000 1.413l5.296 5.294-5.296 5.295a.999.999 0 101.413 1.413l5.295-5.295 5.296 5.295a.999.999 0 001.413 0 1.005 1.005 0 000-1.413l-5.296-5.295h.002z"
                  >
                  </path>
                </svg>
              </span>
            </bolt-icon>
          </replace-with-children>
        </button>
      </bolt-button>
    </li>
  </ul>
</bolt-button-group>
`;
